app/controllers/api/v0/api_controller.rb

```diff
@@ -36,12 +36,13 @@ def error_not_found
       def authenticate!
         if doorkeeper_token
```

Comment:
**non-blocking:**

We could refactor this into a private method if we wanted to make this a tiny bit easier to read (in my opinion).

For example (not tested):
```ruby
def authenticate!
  @user = authenticated_user
  return error_unauthorized unless @user && !@user.banned
end

private

def authenticated_user
  if doorkeeper_token
    return User.find(doorkeeper_token.resource_owner_id)
  elsif request.headers["api-key"]
    return authenticate_with_api_key
  elsif current_user
    return current_user
  end
end
```